src/java.base/share/classes/java/lang/StackWalker.java

```diff
@@ -93,41 +94,44 @@ public final class StackWalker {
      * A {@code StackFrame} object represents a method invocation returned by
      * {@link StackWalker}.
      *
-     * <p> The {@link #getDeclaringClass()} method may be unsupported as determined
-     * by the {@linkplain Option stack walking options} of a {@linkplain
-     * StackWalker stack walker}.
+     * <p> {@linkplain Option <em>Stack walker options</em>} configure the stack
+     * frame information obtained by a {@code StackWalker}.
+     * If the stack walker is configured with {@link Option#DROP_METHOD_INFO
+     * DROP_METHOD_INFO} option, method information such as
+     * the {@linkplain StackFrame#getMethodName() method name},
+     * the {@linkplain StackFrame#getLineNumber() line number},
+     * the {@linkplain StackFrame#getByteCodeIndex() bytecode index}, etc
+     * will be dropped.
+     * If the stack walker is configured with {@link Option#RETAIN_CLASS_REFERENCE
+     * RETAIN_CLASS_REFERENCE} option, the {@link #getDeclaringClass() Class} object
+     * will be retained for access.
      *
      * @since 9
-     * @jvms 2.6
+     * @jvms 2.6 Frames
      */
     public interface StackFrame {
         /**
-         * Gets the <a href="ClassLoader.html#binary-name">binary name</a>
-         * of the declaring class of the method represented by this stack frame.
-         *
-         * @return the binary name of the declaring class of the method
-         *         represented by this stack frame
+         * {@return the <a href="ClassLoader.html#binary-name">binary name</a>
```

Comment:
I think `<a href="ClassLoader.html#binary-name">binary name</a>` can now be replaced with `{@linkplain ClassLoader##binary-name binary name}`. HTH